internal/services/apimanagement/api_management_resource.go

```diff
@@ -713,9 +713,13 @@ func resourceApiManagementServiceCreateUpdate(d *pluginsdk.ResourceData, meta in
 				},
 			}
 
-			if _, err = client.CreateOrUpdate(ctx, id.ResourceGroup, id.ServiceName, params); err != nil {
+			future, err := client.CreateOrUpdate(ctx, id.ResourceGroup, id.ServiceName, params)
+			if err != nil {
 				return fmt.Errorf("recovering %s: %+v", id, err)
 			}
+			if err := future.WaitForCompletionRef(ctx, client.Client); err != nil {
+				return fmt.Errorf("waiting for recovery of %q: %+v", id, err)
+			}
```

Comment:
does this mean that we can remove the ProvisioningState check below?

Reply:
I think we better keep the ProvisioningState check, to avoid something unexpected.

Reply:
in which case there'd no point calling the Future here, since we'd be polling below anyway?